docs/dev/compiler-toolchain/overview.md

```diff
@@ -0,0 +1,68 @@
+# Overview
+
+This section introduces an LLVM-based compiler toolchain for smart contract languages with Ethereum Virtual Machine
+(EVM) support. The toolchain works on top of existing compilers and requires their output, which typically includes
+intermediate representations (IRs), abstract syntax trees (ASTs), and auxiliary contract metadata and documentation.
+::: info
+At the time of writing, we support Solidity and Vyper.
+:::
+
+The toolchain consists of the following:
+
+1. [High-level source code compilers](#high-level-source-code-compilers): `solc` and `vyper`.
+2. [IR compilers, or LLVM front-end](#ir-compilers): `zksolc` and `zkvyper`.
+3. [The LLVM framework](./llvm.md) with a zkEVM back-end which emits the zkEVM text assembly.
+4. [The assembler](#assembler) which produces the zkEVM bytecode from the text assembly.
+5. [Hardhat plugins](#hardhat-plugins) which set up the environment.
+
+## High-level Source Code Compilers
+
+High-level source code compilers are original upstream compilers called by IR compilers as
+child processes to process their output. We use two such compilers at the time of writing:
+
+- [solc](https://github.com/ethereum/solc-bin): the official Solidity compiler. For more info, see [the documentation](https://docs.soliditylang.org/en/latest/).
+- [vyper](https://github.com/vyperlang/vyper/releases): the official Vyper compiler. For more info, see [the documentation](https://docs.vyperlang.org/en/latest/index.html).
```

Comment:
I’d suggest replacing both occurrences of “the documentation” with “its documentation”; using the same phrase twice with different referents evoked a negative linguistic reaction in me.